### PR TITLE
Add poll timeout option to afpacket for non-blocking purpose

### PR DESCRIPTION
**Pull request by safchain**

### CODE REVIEW

**Comment by safchain** [comment]
Hi, 
Any news for this, this can be useful if you want to stop reading even if there is no packet ready.

Thanks

File: afpacket/afpacket.go [hunk 1]
```diff
@@ -39,6 +39,12 @@ import "C"
 var pageSize = int(C.getpagesize())
 var tpacketAlignment = uint(C.TPACKET_ALIGNMENT)
 
+// ErrPoll returns by poll
+var ErrPoll = errors.New("Error condition")
```
Comment (gconnell) [inline afpacket/afpacket.go]: Could you make this error message more descriptive?  Something like "packet poll failed"

Also, please make error messages start with lower-case characters.

File: afpacket/options.go [hunk 2]
```diff
@@ -88,12 +88,17 @@ type OptNumBlocks int
...
+// descriptor to become ready. Specifying a negative value in  time‐out means an infinite timeout.
+type OptPollTimeout time.Duration
+
 // Default constants used by options.
 const (
 	DefaultFrameSize    = 4096                   // Default value for OptFrameSize.
 	DefaultBlockSize    = DefaultFrameSize * 128 // Default value for OptBlockSize.
 	DefaultNumBlocks    = 128                    // Default value for OptNumBlocks.
 	DefaultBlockTimeout = 64 * time.Millisecond  // Default value for OptBlockTimeout.
+	DefaultPollTimeout  = -1 * time.Millisecond  // Default value for OptPollTimeout.
```
Comment (gconnell) [inline afpacket/options.go]: add to comment:  "This blocks forever"

File: afpacket/afpacket.go (hunk 1 above)
Comment (safchain) [inline afpacket/afpacket.go]: Done

File: afpacket/options.go (hunk 2 above)
Comment (safchain) [inline afpacket/options.go]: Done

**Comment by gconnell** [comment]
Thanks!

**Pull request merged**